### PR TITLE
Improvements: Sharepoint OAuth, docs improvements, code clarifications

### PR DESCRIPTION
**Pull request by tianjing-li**
### What's being changed:

Sharepoint connector:
- Adds OAuth as an auth type (keeps existing application auth)
- Improves README, documents both types of auth
- Clarifies client and provider code - notably removing list items (drive items take care of files, so this was unnecessary)

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):

- Tested on local setup, and registering in prod for application/oauth

image

### CODE REVIEW

File: sharepoint/provider/client.py [hunk 1]
```diff
@@ -1,141 +1,126 @@
...
             )
+            if "access_token" not in token_response:
+                raise UpstreamProviderError(
+                    "Error while retrieving access token from Microsoft Graph API"
+                )
+            self.access_token = token_response["access_token"]
         except Exception as e:
             raise UpstreamProviderError(
-                f"Error while initializing Sharepoint client: {str(e)}"
+                f"Error while initializing Teams client: {str(e)}"
```
Comment (EugeneLightsOn) [inline sharepoint/provider/client.py:47]: Small typo ->  Sharepoint client

File: sharepoint/provider/client.py [hunk 2]
```diff
@@ -1,141 +1,126 @@
...
             )
 
-        self.search_limit = search_limit
+    def set_user_access_token(self, token):
+        self.access_token = token
+        self.headers = {"Authorization": f"Bearer {self.access_token}"}
```
Comment (EugeneLightsOn) [inline sharepoint/provider/client.py:53]: self.headers gets set here, but in set_app_access_token() it doesn't, can we init it in the __init__ and add that to the app auth and then everywhere the headers are needed you can use self.headers instead?

File: sharepoint/provider/provider.py [hunk 3]
```diff
@@ -51,48 +50,27 @@ def search(query):
...
     return results
 
 
 def collect_items(sharepoint_client, hits):
     # Gather data
     drive_items = []
-    list_items = []
     for hit in hits:
-        if hit["resource"]["@odata.type"] == MicrosoftDataType.DRIVE_ITEM.value:
+        if hit["resource"]["@odata.type"] == "#microsoft.graph.driveItem":
```
Comment (EugeneLightsOn) [inline sharepoint/provider/provider.py:60]: Let's set the class constant for '#microsoft.graph.driveItem' and remove the enums and consts files as they are no longer used.

**Review by EugeneLightsOn — COMMENTED** [review]
Looks good to me, just couple of comments

File: sharepoint/provider/provider.py (hunk 3 above)
Comment (tianjing-li) [reply sharepoint/provider/provider.py]: Done

File: sharepoint/provider/client.py (hunk 2 above)
Comment (tianjing-li) [reply sharepoint/provider/client.py:55]: Done

**Review by EugeneLightsOn — APPROVED** [review]
Thank you

**Review by daniel-cohere — APPROVED** [review]

**Pull request merged**